File: app/utils/dawproject_python/utility.py
from typing import Set

# Assuming the following classes and enums are already defined in your project
from classes.track import Track
from classes.channel import Channel
from classes.realParameter import RealParameter
from classes.audio import Audio
from classes.fileReference import FileReference
from classes.warp import Warp
from classes.clip import Clip
from classes.clips import Clips
from classes.timeUnit import TimeUnit
from classes.contentType import ContentType
from classes.mixerRole import MixerRole
from classes.unit import Unit
from classes.timeline import Timeline


class Utility:
    @staticmethod
    def create_track(
        name: str,
        content_types: Set[ContentType],
        mixer_role: MixerRole,
        volume: float,
        pan: float,
    ) -> Track:
        track_channel = Channel(
            volume=RealParameter(value=volume, unit=Unit(value="linear")),
            pan=RealParameter(value=pan, unit=Unit(value="normalized")),
            role=mixer_role,
        )
        track = Track(name=name, channel=track_channel, content_type=content_types, loaded=True)
        return track

    @staticmethod
    def create_audio(
        relative_path: str, sample_rate: int, channels: int, duration: float
    ) -> Audio:
        audio = Audio(
            time_unit=TimeUnit(value="seconds"),
            file=FileReference(path=relative_path, external=False),
            sample_rate=sample_rate,
            channels=channels,
            duration=duration,
        )
        return audio

    @staticmethod
    def create_warp(time: float, content_time: float) -> Warp:
        warp = Warp(time=time, content_time=content_time)
        return warp

    @staticmethod
    def create_clip(content: Timeline, time: float, duration: float) -> Clip:
        clip = Clip(content=content, time=time, duration=duration)
        return clip

    @staticmethod
    def create_clips(*clips: Clip) -> Clips:
        clips_timeline = Clips(clips=list(clips))
        return clips_timeline

File: app/utils/dawproject-decorators/utility.ts
import { Channel } from "./channel";
import { ContentType } from "./contentType";
import { FileReference } from "./fileReference";
import { MixerRole } from "./mixerRole";
import { RealParameter } from "./realParameter";
// Needed for accessing static ID
import { Audio } from "./timeline/audio";
import { Clip } from "./timeline/clip";
import { Clips } from "./timeline/clips";
import { TimeUnit } from "./timeline/timeUnit";
import { Warp } from "./timeline/warp";
import { Track } from "./track";
import { Unit } from "./unit";

// Needed for accessing static ID

export class Utility {
  public static createTrack(
    name: string,
    contentTypes: Set<ContentType>,
    mixerRole: MixerRole,
    volume: number,
    pan: number
  ): Track {
    const track = new Track();
    track.channel = new Channel();
    track.name = name;
    const volumeParameter = new RealParameter(Unit.linear);
    volumeParameter.value = volume;
    track.channel.volume = volumeParameter;

    const panParameter = new RealParameter(Unit.normalized);
    panParameter.value = pan;
    track.channel.pan = panParameter;

    track.contentType = Array.from(contentTypes);
    track.channel.role = mixerRole;

    return track;
  }

  public static createAudio(
    relativePath: string,
    sampleRate: number,
    channels: number,
    duration: number
  ): Audio {
    const fileReference = new FileReference(relativePath, false);
    const audio = new Audio(fileReference, sampleRate, channels, duration);
    audio.timeUnit = TimeUnit.seconds;
    return audio;
  }

  public static createWarp(time: number, contentTime: number): Warp {
    return new Warp(time, contentTime);
  }

  public static createClip(content: any, time: number, duration: number): Clip {
    const clip = new Clip(time);
    clip.content = content;
    clip.duration = duration;
    return clip;
  }

  public static createClips(...clips: Clip[]): Clips {
    const timeline = new Clips();
    timeline.clips = clips;
    return timeline;
  }
}
